options/posix/generic/lookup.cpp

```diff
@@ -215,10 +234,27 @@ int lookup_addr_dns(frg::span<char> name, frg::array<uint8_t, 16> &addr, int fam
 	// for dns
 	sin.sin_port = htons(53);
 
-	auto nameserver = get_nameserver();
-	if (!inet_aton(nameserver ? nameserver->name.data() : "127.0.0.1", &sin.sin_addr)) {
-		mlibc::infoLogger() << "lookup_name_dns(): inet_aton() failed!" << frg::endlog;
-		return -EAI_SYSTEM;
+	int nameserver_idx = 0;
+	while (true) {
```

Comment:
Also this while loop doesn't need to be a ``while(true)`` loop, you can also write ``while(nameserver = get_nameserver())``.